README.md

```diff
@@ -48,33 +65,470 @@ cache read/white process, etc. Command handler can be sync and publish the
 of output processing.
 
+## The flow
+
+### Command execution
+
+The first step is to declare a command. Your command type has to conform to
+`Aftermath.Command` protocol and the `Output` type must be implicitly specified.
+
+Let's say we want to fetch a list of books from some untrusted resource and
+correct typos in titles and author names 🤓.
+
+```swift
+// This is our model we are going to work with.
+struct Book {
+  let id: Int
+  let title: String
+  let author: String
+}
+
+struct BookListCommand: Command {
+  // Result of this command will be a list of books.
+  typealias Output = [Book]
+}
+
+struct BookUpdateCommand: Command {
+  // Result of this command will be an updated book.
+  typealias Output = Book
+
+  // Let's pass the entire model to the command to simplify this example.
+  // Ideally we wouldn't do that because a command is supposed to be as simple
+  // as possible, only with attributes that are needed for handler.
+  let book: Book
+}
+```
+
+**Note** that any type can play a role of `Output`, so if we want to add a date
+to our `BookUpdateCommand` it could look like the following:
+
+```swift
+typealias Output = (Book, NSDate)
+```
+
+In order to execute a command you have to conform to `CommandProducer` protocol:
+
+```swift
+class ViewController: UITableViewController, CommandProducer {
+
+  // Fetch a list of books.
+  func load() {
+    execute(command: BookListCommand())
+  }
+
+  // Update a single book with corrected title and/or author name.
+  func update(book: Book) {
+    execute(command: BookUpdateCommand(book: book))
+  }
+}
+```
+
+### Command handling
+
+Command is an intention that needs to be translated into action by handler.
+Command handler is responsible for publishing events to notify about results of
+the operation it performs. Command handler type has to conform to
+`Aftermath.CommandHandler` protocol that needs to know about the command type
+it will work with:
+
+```swift
+struct BookListCommandHandler: CommandHandler {
+
+  func handle(command: BookListCommand) throws -> Event<BooksCommand> {
+    // Start network request to fetch data.
+    fetchBooks { books, error in
+      if let error = error {
+        // Publish error.
+        self.publish(error: error)
+        return
+      }
+
+      // Publish fetched data.
+      self.publish(data: books)
+    }
+
+    // Load data from local database/cache.
+    let localBooks = loadLocalBooks()
+
+    // If the list is empty let the listeners know that operation is in the process.
+    return Book.list.isEmpty ? Event.Progress : Event.Data(localBooks)
+  }
+}
+```
+
+**Note** that every command handler needs to be registered on
+[Aftermath Engine](#engine).
+
+```swift
+Engine.sharedInstance.use(BookListCommandHandler())
+```
+
+### Reacting to events
+
+The last step, but not the least, is to react to events published by command
+handlers. Just conform to `ReactionProducer` protocol, implement reaction
+behavior and you're ready to go:
```

Comment:
From:
> The last step, but not the least, is to react to events published by command
handlers. Just conform to `ReactionProducer` protocol, implement reaction
behavior and you're ready to go:

To:
> The last step, but not the least, is to react to events published by the command
handlers. Just conform to `ReactionProducer` protocol, implement reaction
behavior and you're ready to go: